habitat/plan.sh

```diff
@@ -33,17 +34,9 @@ pkg_deps=(
   core/libarchive
 )
 
-pkg_svc_user=root
-
-do_before() {
-  do_default_before
-  update_pkg_version
-}
-
 do_download() {
-  # Instead of downloading, build a gem based on the source in src/
-  cd $PLAN_CONTEXT/..
```

Comment:
As a best practice, we /shouldn't/ change the directory inside the callbacks since that position will be persistent throughout the build process of the package and latter other callbacks or habitat internals might conflict with that change of state, instead we should 1) execute commands that depend on a specific path inside a sub-sells or 2) use `pushd` and `popd` to move to a directory and get back where we were before. 

Number 1) is more desired but you could use them both. 

Here we actually don't even need to change the directory, so we should try to avoid it.